examples/filtering/send_with_filtering.rs

```diff
@@ -0,0 +1,122 @@
+/* Send 100 messages with filter of Region "California" and 100 messages with filter "Texas"
+Filters are specified in the application_properties of the messages using the custom field "Region"
+in the filter_value_extractor callback
+The main thread wait on a condition variable until all the messages have been confirmed */
+
+use rabbitmq_stream_client::error::StreamCreateError;
+use rabbitmq_stream_client::types::ResponseCode;
+use rabbitmq_stream_client::types::{ByteCapacity, Message};
+use rabbitmq_stream_client::Environment;
+use std::convert::TryInto;
+use std::sync::atomic::{AtomicU32, Ordering};
+use std::sync::Arc;
+use tokio::sync::Notify;
+
+// This callback instruct the Producer on what filter we want to apply
+// In this case we are returning the value of the application_property "region" value
+fn filter_value_extractor(message: &Message) -> String {
+    message
+        .application_properties()
+        .unwrap()
+        .get("region")
+        .unwrap()
+        .clone()
+        .try_into()
+        .unwrap()
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let confirmed_messages = Arc::new(AtomicU32::new(0));
+    let notify_on_send = Arc::new(Notify::new());
+    let stream = "test_stream_filtering";
+
+    let environment = Environment::builder()
+        .host("localhost")
+        .port(5552)
+        .build()
+        .await?;
+
+    let message_count = 200;
+    let create_response = environment
+        .stream_creator()
+        .max_length(ByteCapacity::GB(5))
+        .create(&stream)
+        .await;
+
+    if let Err(e) = create_response {
+        if let StreamCreateError::Create { stream, status } = e {
+            match status {
+                // we can ignore this error because the stream already exists
+                ResponseCode::StreamAlreadyExists => {}
+                err => {
+                    println!("Error creating stream: {:?} {:?}", stream, err);
+                }
+            }
+        }
+    }
+
+    let mut producer = environment
+        .producer()
+        .name("test_producer")
```

Comment:
putting a name enables the deduplcation by default. For the standard examples I'd void it